k8s/sonarqube/README.md

```diff
@@ -0,0 +1,439 @@
+[Stackdriver Metrics Explorer](https://cloud.google.com/monitoring/charts/metrics-explorer).
+
+Each metric of the application will have a name starting with the application's name
+(matching the variable `APP_INSTANCE_NAME` described above).
+
+The exporting option might not be available for GKE on-prem clusters.
+
+> Note: Please be aware that Stackdriver has [quotas](https://cloud.google.com/monitoring/quotas)
+for the number of custom metrics created in a single GCP project. If the quota is met,
+additional metrics will not be accepted by Stackdriver, which might cause that some metrics
+from your application might not show up in the Stackdriver's Metrics Explorer.
+
+Existing metric descriptors can be removed through
+[Stackdriver's REST API](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors/delete).
+
+# Scaling
+
+SonarQube Community Edition doest not support scaling.
+
+
+# Backup and restore
+[From official documentation](https://docs.sonarqube.org/7.6/architecture/architecture-integration/)
+The SonarQube Platform is made of 4 components:
+
+- One SonarQube Server starting 3 main processes:
+
+   - Web Server for developers, managers to browse quality snapshots and configure the SonarQube instance
+   - Search Server based on Elasticsearch to back searches from the UI
+   - Compute Engine Server in charge of processing code analysis reports and saving them in the SonarQube Database
+
+- One SonarQube Database to store:
+
+   - the configuration of the SonarQube instance (security, plugins settings, etc.)
+   - the quality snapshots of projects, views, etc.
+
+- Multiple SonarQube Plugins installed on the server, possibly including language, SCM, integration, authentication, and governance plugins
+
+- One or more SonarScanners running on your Build / Continuous Integration Servers to analyze projects
+
+For our application database the most important place, `plugins` and `data` folders stored on PVC (Persistent Volume Claim).
+## Backing up plugin and data
+This shell script will create copy of plugins folder in current folder
+```shell
+kubectl --namespace $NAMESPACE cp $(kubectl -n$NAMESPACE get pod -oname | \
+                sed -n /\\/$APP_INSTANCE_NAME-sonarqube/s.pods\\?/..p):/opt/sonarqube/extensions/ ./
+```
+This shell script will create copy of content from folder `data` in current folder:
+```shell
+kubectl --namespace $NAMESPACE cp $(kubectl -n$NAMESPACE get pod -oname | \
+                sed -n /\\/$APP_INSTANCE_NAME-sonarqube/s.pods\\?/..p):/opt/sonarqube/data ./
+```
+
+## Backing up PostgreSQL
+This shell script will create `postgresql-backup.sql` dump of all DB in PostgreSQL.
+```shell
+kubectl --namespace $NAMESPACE exec -t \
+	$(kubectl -n$NAMESPACE get pod -oname | \
+		sed -n /\\/$APP_INSTANCE_NAME-postgresql-deployment/s.pods\\?/..p) \
+	-c postgresql-server \
+	-- pg_dumpall -c -U postgres > backup.sql
+```
+
+## Restoring your PostgreSQL
```

Comment:
```suggestion
## Restoring your PostgreSQL

```

Could you apply this pattern to all headers in this readme?

https://github.com/google/styleguide/blob/gh-pages/docguide/style.md#headings